queue/src/main/java/com/bazaarvoice/emodb/queue/core/AbstractQueueService.java

```diff
@@ -47,6 +47,8 @@
     private final JobType<MoveQueueRequest, MoveQueueResult> _moveQueueJobType;
     private final LoadingCache<SizeCacheKey, Map.Entry<Long, Long>> _queueSizeCache;
 
+    public static final int MAX_MESSAGE_SIZE_IN_BYTES = 30 * 1024;
```

Comment:
if we want to, it's only relevant to teh queue service so I thought this placement might be better but we can move it

Reply:
I would argue that it is relevant to any service that uses slabs, including the the databus. Although the databus doesn't have this issue right now, putting this protection at the DAO level could protect us, especially because any event larger than MAX_SLAB_SIZE will not cause some very dangerous behavior.

Reply:
Well Ben everything else in Constants.java is used in the ChannelAllocationState and other slab allocation/management classes, but MAX_MESSAGE_SIZE isn't it strictly applies to queues. Not something I care to fall on my sword about, but nothing in slab management cares about it.

The one vulnerability we have here is that there is nothing in the channel allocation state to make sure an event size is > the size of a slab, which will cause it to malfunction. Now that can't happen from a queueing request because the queue API checks the message size, but if a databus or some future non-queue request could come in that was bigger than a slab then it would break. However, I don't think that is very likely so it's probably not worth messing with. Maybe I could add a comment about it though. Oh and I have to remove those vestigial functions!

Reply:
I would rather have the check in the DAO, as the limitation is actually one of the DAO implementation, not the queueing service itself.

Reply:
Hmmm I agree the DAO should have a check but you can't use IllegalArgumentException down there presently, so throwing an exception will cause a 500. Having the check in the queueing service is nice because you can just say it is an illegal argument (which it is, since we are limiting it regardless of slabs etc) and that gives a more meaningful exception. I would be OK with the DAO throwing a 500 because nothing should ever get in there that is beyond the max document size we allow to be enqueued.

Reply:
Why can't you have IllegalArgumentException down there?

Reply:
Oh actually I can, I was thinking of another place where I tried to put an IllegalArgumentException and it complained about it not being in the method signature etc, looks like here it will work. Are you saying we should remove teh check from the Queueing API as well?

Reply:
There is no harm in it being in both places, my feeling was just that it makes more sense from the DAO to throw the exception, because it is the DAO implementation that needs to be protected from large messaged, rather than a limitation with the api. Additionally, if we ever made a change to the databus which led to larger message sizes, we are protected by the DAO's checks.